warzone/Player.cpp

```diff
@@ -1,11 +1,95 @@
 #include "Player.h"
 
-Player::Player()
-{
+
+Player::Player() {  //default constructor
+	this->playerTerritories = new vector<Territory*>();
 	this->playerOrdersList = new OrdersList();
 }
 
-OrdersList* Player::getOrdersList()
+
+Player::Player(vector<Territory*>* territoriesToAdd, OrdersList* playerList, Deck* deckToTakeFrom) { //Constructor with all parameters
+	this->playerTerritories = territoriesToAdd;
+	this->playerOrdersList = playerList;
+	this->playerHand = new Hand(deckToTakeFrom, this);
+}
+
+Player::~Player() {  //destructor 
+	delete playerTerritories;
+	delete playerHand;
+	delete playerOrdersList;
+}
+
+
+Player* Player::operator=(const Player & one)  //assignment operator
+{
+	return new Player(one);
+}
+
+Player::Player(const Player& player)  //copy constructor
+{
+	for (Territory* t : *player.playerTerritories)
+	{
+		Territory temp = new Territory(t);
+		this->playerTerritories->push_back(t);    //copies each territory in the incoming player's territories 
+	}
```

Comment:
Shouldn't you be inserting `temp`, not `t`, into the array?

Reply:
ouch, good catch!